lighthouse-core/audits/lcp-lazy-loaded.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit.js');
+const i18n = require('../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Descriptive title of a diagnostic audit that provides */
+  title: 'Largest Contentful Paint element was lazy-loaded',
+  /** Description of a Lighthouse audit that tells */
+  description: 'Consider to remove lazy loading for largest contentful paint element.',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LargestContentfulPaintLazyLoaded extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'lcp-lazy-loaded',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      supportedModes: ['navigation'],
+      requiredArtifacts: ['TraceElements', 'ViewportDimensions', 'ImageElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts.ImageElement} image
+   * @param {LH.Artifacts.ViewportDimensions} viewportDimensions
+   * @return {boolean}
+   */
+  static isImageInViewport(image, viewportDimensions) {
+    const imageTop = image.clientRect.top;
+    const viewportHeight = viewportDimensions.innerHeight;
+    return imageTop < viewportHeight;
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    const images = artifacts.ImageElements;
+    const lazyLoadedImages = images.filter(
```

Comment:
We can be a little more specific with our audit results if we avoid filtering to lazy from the start.

something more like

```js
const lcpElmentImage = images.find(elem => elem.node.devtoolsNodePath === lcpElement.node.devtoolsNodePath);

if (!lcpElementImage || !this.isImageInViewport(lcpElementImage, artifacts.ViewportDimensions)) {
  return {score: 1, notApplicable: true};
}

const details = /* as-is ... */ {};
return {
  score: lcpElementImage.loading === 'lazy' ? 0 : 1, 
  details,
}
```

This will provide clearer not applicable and passing states :)